src/pip/_internal/network/lazy_wheel.py

```diff
@@ -131,15 +183,16 @@ def writable(self) -> bool:
         """Return False."""
         return False
 
-    def __enter__(self) -> "LazyZipOverHTTP":
+    def __enter__(self) -> LazyZipOverHTTP:
         self._file.__enter__()
         return self
 
-    def __exit__(self, *exc: Any) -> None:
-        self._file.__exit__(*exc)
+    def __exit__(self, *exc: Any) -> bool | None:
+        print(self._request_count, "requests to fetch metadata from", self._url[107:])
```

Comment:
Stray debugging code?

Reply:
Which test are you referring to here?

Reply:
Could we use `log.debug()` like we do elsewhere in this change?